pkg/oc/cli/importimage/importimage.go

```diff
@@ -481,23 +532,34 @@ func (o *ImportImageOptions) newImageStreamImportAll(stream *imageapi.ImageStrea
 	return isi
 }
 
-func (o *ImportImageOptions) newImageStreamImportTags(stream *imageapi.ImageStream, tags map[string]string) *imageapi.ImageStreamImport {
+func (o *ImportImageOptions) newImageStreamImportTags(stream *imageapiv1.ImageStream, tags map[string]string) *imageapiv1.ImageStreamImport {
 	isi, streamInsecure := o.newImageStreamImport(stream)
 	for tag, from := range tags {
 		insecure := streamInsecure
 		scheduled := o.Scheduled
-		oldTag, ok := stream.Spec.Tags[tag]
-		if ok {
+
+		oldTagFound := false
+		var oldTag imageapiv1.TagReference
+		for _, t := range stream.Spec.Tags {
```

Comment:
You're iterating over this second time so far, create a helper and use in both places.